### PR TITLE
Send notification in case AVS crashes (in most cases)

### PR DESCRIPTION
**Pull request by ptare**
If something failed while constructing the object VotingSystem or
sending a notification, then we won't be able to alert for that.

This PR should handle the case where the call to `batchCommit` or `batchReveal` fails.

Signed-off-by: Prasad Tare <prasad.tare@gmail.com>

### CODE REVIEW

File: core/test/scripts/Voting.js [hunk 1]
```diff
@@ -195,6 +195,33 @@ contract("scripts/Voting.js", function(accounts) {
     assert.equal((await voting.getPrice(identifier, time)).toString(), web3.utils.toWei("9155.05"));
   });
 
+  it("Notification on crash", async function() {
+    // Force a Solidity revert by returning an invalid hash in the soliditySha3 function.
+    const temp = web3.utils.soliditySha3;
+    web3.utils.soliditySha3 = () => {
+      return "0x0";
+    };
```
Comment (ptare) [inline core/test/scripts/Voting.js]: Somewhat hacky, do you know of a better way?

Comment (mrice32) [reply core/test/scripts/Voting.js]: This seems reasonable to me - it's a test, so IMO hackiness is totally acceptable if it keeps things simple.

File: core/test/scripts/Voting.js [hunk 2]
```diff
@@ -195,6 +195,33 @@ contract("scripts/Voting.js", function(accounts) {
...
+    await voting.requestPrice(identifier, time);
+
+    // Run an iteration, which should crash.
+    const notifier = new MockNotifier();
+    const votingSystem = new VotingScript.VotingSystem(voting, voter, [notifier]);
+    await moveToNextRound(voting);
+    await votingSystem.runIteration();
+
+    // A notification email should be sent.
+    assert.equal(notifier.notificationsSent, 1);
```
Comment (mrice32) [inline core/test/scripts/Voting.js]: Just so I understand, would this test still succeed even if `runIteration` didn't crash? (I think that's fine, just wanted to double check my understanding)

Comment (ptare) [reply core/test/scripts/Voting.js]: Yep. What do you think of this version? Feels somewhat safer to mock on an object we control.

Comment (mrice32) [reply core/test/scripts/Voting.js]: Agreed, this is way better!

**Pull request merged**